### DISCUSSION TITLE
How to print the sources of vector search just like langchain?

### DISCUSSION THREAD

**Discussion by abrahimzaman360** [safevideo/autollm#212] [Q&A] [answered]

**Comment by abrahimzaman360** [comment]
How to print the sources of vector search just like langchain?
I mean from which context it is returning result

**Comment by talhasarit** [comment, marked as answer]
Hi @AbrahimZaman360,

After you get your response from the query engine:

```python
response = query_engine.query(question)
```

you can get the source nodes via `response.get_formatted_sources` method.

```python
print(response.get_formatted_sources(length=42000))
```

will probably return all the source text you have got. `length` parameter is for adjusting the size of the returned source context.

**Comment by abrahimzaman360** [comment]
Thanks 👍🏼 for helping out
